test/common/router/retry_state_impl_test.cc

```diff
@@ -834,21 +834,21 @@ TEST_F(RouterRetryStateImplTest, Backoff) {
   setup(request_headers);
   EXPECT_TRUE(state_->enabled());
 
-  EXPECT_CALL(random_, random()).WillOnce(Return(49));
-  retry_timer_ = new Event::MockTimer(&dispatcher_);
-  EXPECT_CALL(*retry_timer_, enableTimer(std::chrono::milliseconds(24), _));
+  EXPECT_CALL(random_, random()).WillOnce(Return(190));
```

Comment:
qq: why does this test need to change as well as the HDS test? It seems like this fix shouldn't have any behavior change in the normal case?

Reply:
The issue is related to this part of the original implementation that traces back to the first revision of Envoy in github:
uint32_t multiplier = (1 << current_retry_) - 1;

This is not increasing backoff by a factor of 2 each time.  I think that what was really intended was:
uint32_t multiplier = 1 << (current_retry_ - 1);

Current backoff sequence: 1, 3, 7, 15...
New backoff sequence: 1, 2, 4, 8, 16...
The biggest difference is the jump from increase from 1 to 3x instead of 2x at the first step.  Replicating the current behavior is possible, but the code would be slightly more complicated.

Some discussion about it here:
https://github.com/envoyproxy/envoy/pull/3791#discussion_r201557885

Reply:
Ah OK got it, makes sense, thanks.

Reply:
Of course, thanks for digging deeper on unexplained code changes.